// @esri/calcite-components/dist/cjs/en-ca-36fa64d1.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.11.0
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-30d23b46.js');
require('./index-e433dc3a.js');
require('./form-1432ae43.js');
require('./dom-b708de15.js');
require('./guid-bc43d543.js');
require('./resources-a31ffd1a.js');
require('./interactive-79b6c5f7.js');
require('./browser-881f0556.js');
require('./key-a08abeaa.js');
require('./label-81185d54.js');
require('./loadable-47839cdc.js');
require('./locale-a273bc0a.js');
require('./observers-ad39b7aa.js');
require('./focusTrapComponent-8b961e53.js');
require('./math-21cdbe9b.js');
require('./t9n-3fe28893.js');
require('./openCloseComponent-1dd76bff.js');
require('./component-8aad1833.js');

// English (Canada) [en-ca]
var locale = {
  name: 'en-ca',
  weekdays: 'Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday'.split('_'),
  months: 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_'),
  weekdaysShort: 'Sun_Mon_Tue_Wed_Thu_Fri_Sat'.split('_'),
  monthsShort: 'Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec'.split('_'),
  weekdaysMin: 'Su_Mo_Tu_We_Th_Fr_Sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'h:mm A',
    LTS: 'h:mm:ss A',
    L: 'YYYY-MM-DD',
    LL: 'MMMM D, YYYY',
    LLL: 'MMMM D, YYYY h:mm A',
    LLLL: 'dddd, MMMM D, YYYY h:mm A'
  },
  relativeTime: {
    future: 'in %s',
    past: '%s ago',
    s: 'a few seconds',
    m: 'a minute',
    mm: '%d minutes',
    h: 'an hour',
    hh: '%d hours',
    d: 'a day',
    dd: '%d days',
    M: 'a month',
    MM: '%d months',
    y: 'a year',
    yy: '%d years'
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports.default = locale;
