// @esri/calcite-components/dist/cjs/tr-f8f2c671.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.11.0
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-30d23b46.js');
require('./index-e433dc3a.js');
require('./form-1432ae43.js');
require('./dom-b708de15.js');
require('./guid-bc43d543.js');
require('./resources-a31ffd1a.js');
require('./interactive-79b6c5f7.js');
require('./browser-881f0556.js');
require('./key-a08abeaa.js');
require('./label-81185d54.js');
require('./loadable-47839cdc.js');
require('./locale-a273bc0a.js');
require('./observers-ad39b7aa.js');
require('./focusTrapComponent-8b961e53.js');
require('./math-21cdbe9b.js');
require('./t9n-3fe28893.js');
require('./openCloseComponent-1dd76bff.js');
require('./component-8aad1833.js');

// Turkish [tr]
var locale = {
  name: 'tr',
  weekdays: 'Pazar_Pazartesi_Salı_Çarşamba_Perşembe_Cuma_Cumartesi'.split('_'),
  weekdaysShort: 'Paz_Pts_Sal_Çar_Per_Cum_Cts'.split('_'),
  weekdaysMin: 'Pz_Pt_Sa_Ça_Pe_Cu_Ct'.split('_'),
  months: 'Ocak_Şubat_Mart_Nisan_Mayıs_Haziran_Temmuz_Ağustos_Eylül_Ekim_Kasım_Aralık'.split('_'),
  monthsShort: 'Oca_Şub_Mar_Nis_May_Haz_Tem_Ağu_Eyl_Eki_Kas_Ara'.split('_'),
  weekStart: 1,
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY HH:mm',
    LLLL: 'dddd, D MMMM YYYY HH:mm'
  },
  relativeTime: {
    future: '%s sonra',
    past: '%s önce',
    s: 'birkaç saniye',
    m: 'bir dakika',
    mm: '%d dakika',
    h: 'bir saat',
    hh: '%d saat',
    d: 'bir gün',
    dd: '%d gün',
    M: 'bir ay',
    MM: '%d ay',
    y: 'bir yıl',
    yy: '%d yıl'
  },
  ordinal: function ordinal(n) {
    return n + ".";
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports.default = locale;
