// @esri/calcite-components/dist/esm/es-mx-b67051f6.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.11.0
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-17fe7dfa.js';
import './index-5581518c.js';
import './form-aaedb8eb.js';
import './dom-bb94e253.js';
import './guid-b69b57e2.js';
import './resources-53455208.js';
import './interactive-69ec9d65.js';
import './browser-2f0866ce.js';
import './key-4eff881c.js';
import './label-945638ac.js';
import './loadable-e0edfe27.js';
import './locale-e5414778.js';
import './observers-7bce645e.js';
import './focusTrapComponent-cb668640.js';
import './math-63a03baa.js';
import './t9n-4e8ea923.js';
import './openCloseComponent-a2872792.js';
import './component-5f2c22d5.js';

// Spanish (Mexico) [es-mx]
var locale = {
  name: 'es-mx',
  weekdays: 'domingo_lunes_martes_miércoles_jueves_viernes_sábado'.split('_'),
  weekdaysShort: 'dom._lun._mar._mié._jue._vie._sáb.'.split('_'),
  weekdaysMin: 'do_lu_ma_mi_ju_vi_sá'.split('_'),
  months: 'enero_febrero_marzo_abril_mayo_junio_julio_agosto_septiembre_octubre_noviembre_diciembre'.split('_'),
  monthsShort: 'ene_feb_mar_abr_may_jun_jul_ago_sep_oct_nov_dic'.split('_'),
  relativeTime: {
    future: 'en %s',
    past: 'hace %s',
    s: 'unos segundos',
    m: 'un minuto',
    mm: '%d minutos',
    h: 'una hora',
    hh: '%d horas',
    d: 'un día',
    dd: '%d días',
    M: 'un mes',
    MM: '%d meses',
    y: 'un año',
    yy: '%d años'
  },
  ordinal: function ordinal(n) {
    return n + "\xBA";
  },
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'DD/MM/YYYY',
    LL: 'D [de] MMMM [de] YYYY',
    LLL: 'D [de] MMMM [de] YYYY H:mm',
    LLLL: 'dddd, D [de] MMMM [de] YYYY H:mm'
  }
};
dayjs.locale(locale, null, true);

export default locale;
