// @esri/calcite-components/dist/esm/fr-ch-69c37efc.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.11.0
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-17fe7dfa.js';
import './index-5581518c.js';
import './form-aaedb8eb.js';
import './dom-bb94e253.js';
import './guid-b69b57e2.js';
import './resources-53455208.js';
import './interactive-69ec9d65.js';
import './browser-2f0866ce.js';
import './key-4eff881c.js';
import './label-945638ac.js';
import './loadable-e0edfe27.js';
import './locale-e5414778.js';
import './observers-7bce645e.js';
import './focusTrapComponent-cb668640.js';
import './math-63a03baa.js';
import './t9n-4e8ea923.js';
import './openCloseComponent-a2872792.js';
import './component-5f2c22d5.js';

// French (Switzerland) [fr-ch]
var locale = {
  name: 'fr-ch',
  weekdays: 'dimanche_lundi_mardi_mercredi_jeudi_vendredi_samedi'.split('_'),
  months: 'janvier_février_mars_avril_mai_juin_juillet_août_septembre_octobre_novembre_décembre'.split('_'),
  weekStart: 1,
  weekdaysShort: 'dim._lun._mar._mer._jeu._ven._sam.'.split('_'),
  monthsShort: 'janv._févr._mars_avr._mai_juin_juil._août_sept._oct._nov._déc.'.split('_'),
  weekdaysMin: 'di_lu_ma_me_je_ve_sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY HH:mm',
    LLLL: 'dddd D MMMM YYYY HH:mm'
  },
  relativeTime: {
    future: 'dans %s',
    past: 'il y a %s',
    s: 'quelques secondes',
    m: 'une minute',
    mm: '%d minutes',
    h: 'une heure',
    hh: '%d heures',
    d: 'un jour',
    dd: '%d jours',
    M: 'un mois',
    MM: '%d mois',
    y: 'un an',
    yy: '%d ans'
  }
};
dayjs.locale(locale, null, true);

export default locale;
