// @esri/calcite-components/dist/esm/tr-141ac71e.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.11.0
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-17fe7dfa.js';
import './index-5581518c.js';
import './form-aaedb8eb.js';
import './dom-bb94e253.js';
import './guid-b69b57e2.js';
import './resources-53455208.js';
import './interactive-69ec9d65.js';
import './browser-2f0866ce.js';
import './key-4eff881c.js';
import './label-945638ac.js';
import './loadable-e0edfe27.js';
import './locale-e5414778.js';
import './observers-7bce645e.js';
import './focusTrapComponent-cb668640.js';
import './math-63a03baa.js';
import './t9n-4e8ea923.js';
import './openCloseComponent-a2872792.js';
import './component-5f2c22d5.js';

// Turkish [tr]
var locale = {
  name: 'tr',
  weekdays: 'Pazar_Pazartesi_Salı_Çarşamba_Perşembe_Cuma_Cumartesi'.split('_'),
  weekdaysShort: 'Paz_Pts_Sal_Çar_Per_Cum_Cts'.split('_'),
  weekdaysMin: 'Pz_Pt_Sa_Ça_Pe_Cu_Ct'.split('_'),
  months: 'Ocak_Şubat_Mart_Nisan_Mayıs_Haziran_Temmuz_Ağustos_Eylül_Ekim_Kasım_Aralık'.split('_'),
  monthsShort: 'Oca_Şub_Mar_Nis_May_Haz_Tem_Ağu_Eyl_Eki_Kas_Ara'.split('_'),
  weekStart: 1,
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY HH:mm',
    LLLL: 'dddd, D MMMM YYYY HH:mm'
  },
  relativeTime: {
    future: '%s sonra',
    past: '%s önce',
    s: 'birkaç saniye',
    m: 'bir dakika',
    mm: '%d dakika',
    h: 'bir saat',
    hh: '%d saat',
    d: 'bir gün',
    dd: '%d gün',
    M: 'bir ay',
    MM: '%d ay',
    y: 'bir yıl',
    yy: '%d yıl'
  },
  ordinal: function ordinal(n) {
    return n + ".";
  }
};
dayjs.locale(locale, null, true);

export default locale;
